common/json/platform/nlohmann/nlohmann_json.cpp

```diff
@@ -12,24 +12,141 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-#include <iostream>
 #include <common/json.hpp>
+
+#include <fstream>
+#include <string>
 #include <nlohmann/json.hpp>
 
 using njson = nlohmann::json;
+using namespace std;
+
+namespace mender::common::json {
+
+ExpectedJson LoadFromFile(string file_path) {
+	ifstream f(file_path);
+	try {
+		njson parsed = njson::parse(f);
+		Json j = Json(parsed);
+		return ExpectedJson(j);
+	} catch (njson::parse_error &e) {
+		JsonError err = {
+			JsonErrorCode::ParseError, "Failed to parse '" + file_path + "': " + e.what()};
+		return ExpectedJson(err);
+	}
+}
+
+ExpectedJson LoadFromString(string json_str) {
+	try {
+		njson parsed = njson::parse(json_str);
+		Json j = Json(parsed);
+		return ExpectedJson(j);
+	} catch (njson::parse_error &e) {
+		JsonError err = {
+			JsonErrorCode::ParseError, "Failed to parse '''" + json_str + "''': " + e.what()};
+		return ExpectedJson(err);
+	}
+}
+
+string Json::Dump(const int indent) const {
+	return this->n_json.dump(indent);
+}
+
+ExpectedJson Json::Get(const char *child_key) const {
+	bool contains = this->n_json.contains(child_key);
+	if (!contains) {
+		JsonError err = {JsonErrorCode::KeyError, "Key '" + string(child_key) + "' doesn't exist"};
+		return ExpectedJson(err);
+	}
+	try {
+		njson n_json = this->n_json[child_key];
+		Json j = Json(n_json);
+		return ExpectedJson(j);
+	} catch (njson::type_error &e) {
+		JsonError err = {
+			JsonErrorCode::TypeError, "Invalid JSON type to get '" + string(child_key) + "' from"};
+		return ExpectedJson(err);
+	}
+}
 
-namespace json {
+ExpectedJson Json::Get(const size_t idx) const {
+	if (this->n_json.size() <= idx) {
```

Comment:
Question: Should we make an explicit check for whether this is in array? It could help for debugging purposes.

The opposite question applies to the Get() by key.

Reply:
I was thinking I would leave this to nlohmann::json, but it cannot hurt, I guess.